config/config.go

```diff
@@ -32,6 +32,30 @@ type ConfigVar struct {
 	keys            []string
 }
 
+func GetTotalHotReloadableConfigVars() int {
+	configVarLock.RLock()
+	defer configVarLock.RUnlock()
+
+	total := 0
+	for _, configVars := range hotReloadableConfig {
+		total += len(configVars)
+	}
+
+	return total
+}
+
+func GetTotalNonHotReloadableConfigVars() int {
+	configVarLock.RLock()
+	defer configVarLock.RUnlock()
+
+	total := 0
+	for _, configVars := range nonHotReloadableConfig {
```

Comment:
It seems to me that the only reason why we retain pointers to non-hot-reloadable config options is for doing `exit(1)` when `RESTART_ON_CONFIG_CHANGE=true` and a configuration update affects some non hot reloadable config option pointer.

It also seems to me that `RESTART_ON_CONFIG_CHANGE` is a feature that we don't use, ever.

Based on the above, I would propose to change the scope of this PR a bit: 
Remove `nonHotReloadableConfig`, drop support for `RESTART_ON_CONFIG_CHANGE` and we won't have to worry about properly closing our `FileManager`s anymore :)

Reply:
We introduced this `RESTART_ON_CONFIG_CHANGE` because previously there were cases when people changed the config and assumed it would reload automatically, So to avoid the confusion we introduced this change but as you said this env isn't passed now via operator

Reply:
If we need to retain the `RESTART_ON_CONFIG_CHANGE` feature we don't need pointers, just the config keys of registered non-hot-reloadable options. But I don't see an actual need for retaining this feature

Reply:
Agreed we don't need to save pointers. 

Only other advantage I can think of nonHotReloadableConfig map is, if we ever want to inspect what values the variables are currently set to (say via rudder-cli), we can just print the value pointer are pointing to.

We can debate on this and take up in another pr. For now, resolving this. :)

Reply:
Keeping the map but replacing the pointer with an empty struct should be enough to simplify this and we wouldn't need to introduce a `Close` method in filemanager anymore. You might have introduced `Close` where necessary now, but any further usage of filemanager should also deal with closing it. A design where you don't having to remember to use close before disposing the filemanager will always better than a design where you have.
That being said, I would debate that such a simplification is desirable & safe, thus we should do it in the context of this PR already.
If everybody else feels differently though, I agree to take it up in another PR :)

Reply:
Just realised, timeout variables were not hotreloadable. 

It is desirable to make them hotreloadable. But instead of adding Close overhead, fetching the value from config will make things simple.

Reply:
During testing, there occasions where we want to override the timeout manually, e.g. [here](https://github.com/rudderlabs/rudder-server/blob/e242c7d5d94ecc903f2565639a1b43b4a3d31ffa/warehouse/configuration_testing/helper.go#L62). Therefore, config value should be used only if timeout is not manually set.